### PR TITLE
Improve install script

### PR DESCRIPTION
**Pull request by Itai-Nelken**
`\e[1m` = bold text.
`\e[31m` = red text.
`\e[1;31m` = red + bold text.
`\e[0m` = reset.
[bash color formmating (also works with printf in C)](https://misc.flogisoft.com/bash/tip_colors_and_formatting#reset)

### CODE REVIEW

**Comment by CleanMachine1** [comment]
Wow, thanks for all the work you have put into this script, its more code than DollarSkip!


I have a few errors in this so I have added  a suggestion (in 5 mins)

File: install.sh [hunk 1]
```diff
@@ -1,20 +1,57 @@
...
 
-git clone https://github.com/CleanMachine1/DollarSkip
-cd DollarSkip
-make 
-sudo make install
+if [[ ! -d DollarSkip ]]; then
+   git clone https://github.com/CleanMachine1/DollarSkip.git || error "Failed to clone DollarSkip repository!"
+else
+   while true; do
+      echo -e "\e[1m\"DollarSkip\" folder already exists, update it (git pull)?\e[0m"
```
Comment (CleanMachine1) [inline install.sh:38]:
```suggestion
      echo -e "\e[1m\"DollarSkip\" folder already exists, would you like to update it (git pull)?\e[0m"
```

File: install.sh [hunk 2]
```diff
@@ -1,20 +1,57 @@
...
+      read -r answer
+      if [[ "$answer" =~ [yY] ]]; then
+         cd DollarSkip || error "Failed to enter DollarSKip folder!"
```
Comment (CleanMachine1) [inline install.sh:41]:
```suggestion
         cd DollarSkip || error "Failed to enter DollarSkip folder!"
```

**Comment by CleanMachine1** [comment]
> curl -sSl https://raw.githubusercontent.com/Itai-Nelken/DollarSkip/patch-1/install.sh | bash
"DollarSkip" folder already exists, update it (git pull)?
make: *** No targets specified and no makefile found. Stop.
Failed to run "make"!


Gave me this instantly, without my input

**Comment by CleanMachine1** [comment]
Ok hi, we have some debugging to do I think

**Comment by Itai-Nelken** [comment]
The problem is piping the script to bash...
There is no way I know of to ask a question in bash when piping the script to it.
Sorry I didn't respobd earlier, I was on a trip.

**Comment by Itai-Nelken** [comment]
Tge script works perfectly fine for me when run locally.

**Comment by CleanMachine1** [comment]
cd DollarSkip || error "Failed to enter DollarSkip folder!":49

Since I already am cd'ed in into it because, it done it on line 42

File: install.sh [hunk 3]
```diff
@@ -1,20 +1,57 @@
...
+      read -r answer
+      if [[ "$answer" =~ [yY] ]]; then
+         cd DollarSkip || error "Failed to enter DollarSkip folder!"
+         git pull || error "Failed to run \"git pull\"!"
+         break
+      else
+         break
+      fi
+   done
+fi
```
Comment (CleanMachine1) [inline install.sh:48]:
```suggestion
fi
if [[ $(pwd | grep DollarSkip -c) = 0 ]]; then
   cd DollarSkip || error "test"
fi
```

Comment (CleanMachine1) [reply install.sh:48]: This fixes the issues

**Comment by CleanMachine1** [comment]
Maybe we should just pull regardless, seems no harm

**Comment by Itai-Nelken** [comment]
I fixed the syntax error and the double `cd`, now the script runs when not piped into bash:
```bash
itai@pinebookpro:~/Desktop/git/DollarSkip$ bash install.sh 
Cloning into 'DollarSkip'...
remote: Enumerating objects: 127, done.
remote: Counting objects: 100% (127/127), done.
remote: Compressing objects: 100% (84/84), done.
remote: Total 127 (delta 59), reused 87 (delta 33), pack-reused 0
Receiving objects: 100% (127/127), 38.08 KiB | 251.00 KiB/s, done.
Resolving deltas: 100% (59/59), done.
gcc dollarskip.c -o temp
cp temp /usr/bin/\$
rm temp
If you see 'It Works!', then it works!
It Works!
itai@pinebookpro:~/Desktop/git/DollarSkip$ bash install.sh 
"DollarSkip" folder already exists, would you like to update it (git pull) [y/n]?y
Already up to date.
gcc dollarskip.c -o temp
cp temp /usr/bin/\$
rm temp
If you see 'It Works!', then it works!
It Works!
itai@pinebookpro:~/Desktop/git/DollarSkip$ 
```

**Comment by Itai-Nelken** [comment]
> Maybe we should just pull regardless, seems no harm

good idea. I added the check and question in case someone made changes and they got deleted when running `git pull`, but thinking about it again I realized that git will refuse to pull if you have uncommitted changes.
so I will just git pull without asking the user, is that ok?

File: install.sh (hunk 3 above)
Comment (Itai-Nelken) [reply install.sh:50]: I fixed them in a way I feel more comfortable with. feel free to change to this if you think its better.

**Comment by CleanMachine1** [comment]
There shouln't be user changes, the outputted binaries are moved/deleted with the clean

**Comment by CleanMachine1** [comment]
We have to be able to pipe it to bash, from curl since otherwise it probably wont be a 1-liner

**Comment by Itai-Nelken** [comment]
we also have to remove the questions for installing make and git (if they are not installed) if we pipe it to bash.
also its better to use `wget` as curl is not installed by default on some systems.

**Comment by CleanMachine1** [comment]
I'll do some testing in around an hour

**Comment by Itai-Nelken** [comment]
the install script is working fine for me when piping to bash using this command: `wget -qO- https://raw.githubusercontent.com/Itai-Nelken/DollarSkip/patch-1/install.sh | bash`
it now searches for `gcc` as well and works with `pacman` and `apt`.

File: install.sh [hunk 4]
```diff
@@ -1,20 +1,62 @@
...
+
+if [[ ! -d DollarSkip ]]; then
+	git clone https://github.com/CleanMachine1/DollarSkip.git || error "Failed to clone DollarSkip repository!"
+	cd DollarSkip || error "Failed to enter DollarSkip folder!"
+else
+	cd DollarSkip || error "Failed to enter DollarSkip folder!"
+	git pull || error "Failed to update repository!"
+fi
+make || error "Failed to run \"make\"!"
```
Comment (CleanMachine1) [inline install.sh:55]:
```suggestion
clear -x 
make || error "Failed to run \"make\"!"
```

Comment (CleanMachine1) [reply install.sh:55]: Added clear -x

**Comment by CleanMachine1** [comment]
I didn't like the fact that like 80% of the code was shell, so I have added a removal of it here https://github.com/CleanMachine1/DollarSkip/commit/2cc9fa917841a46dadab76ccd84f86260db1cf8b

**Pull request merged**